clang/test/CodeGenSYCL/intel-fpga-loops.cpp

```diff
@@ -1,55 +1,35 @@
 // RUN: %clang_cc1 -triple spir64-unknown-unknown-sycldevice -disable-llvm-passes -fsycl-is-device -emit-llvm %s -o - | FileCheck %s
 
-// CHECK: br label %for.cond, !llvm.loop ![[MD_A:[0-9]+]]
-// CHECK: br label %for.cond, !llvm.loop ![[MD_B:[0-9]+]]
-// CHECK: br label %for.cond, !llvm.loop ![[MD_C:[0-9]+]]
-// CHECK: br label %for.cond2, !llvm.loop ![[MD_D:[0-9]+]]
-// CHECK: br label %for.cond,  !llvm.loop ![[MD_E:[0-9]+]]
-// CHECK: br label %for.cond2, !llvm.loop ![[MD_F:[0-9]+]]
+// CHECK: br label %for.cond,  !llvm.loop ![[MD_II:[0-9]+]]
+// CHECK: br label %for.cond2, !llvm.loop ![[MD_II_2:[0-9]+]]
+// CHECK: br label %for.cond,  !llvm.loop ![[MD_MC:[0-9]+]]
+// CHECK: br label %for.cond2, !llvm.loop ![[MD_MC_2:[0-9]+]]
 
-// CHECK: ![[MD_A]] = distinct !{![[MD_A]], ![[MD_ii:[0-9]+]]}
-// CHECK-NEXT: ![[MD_ii]] = !{!"llvm.loop.ii.count", i32 2}
-void goo() {
-  int a[10];
-  [[intelfpga::ii(2)]]
-  for (int i = 0; i != 10; ++i)
-    a[i] = 0;
-}
-
-// CHECK: ![[MD_B]] = distinct !{![[MD_B]], ![[MD_max_concurrency:[0-9]+]]}
-// CHECK-NEXT: ![[MD_max_concurrency]] = !{!"llvm.loop.max_concurrency.count", i32 2}
-void zoo() {
-  int a[10];
-  [[intelfpga::max_concurrency(2)]]
-  for (int i = 0; i != 10; ++i)
-    a[i] = 0;
-}
-
-// CHECK: ![[MD_C]] = distinct !{![[MD_C]], ![[MD_ii_2:[0-9]+]]}
-// CHECK-NEXT: ![[MD_ii_2]] = !{!"llvm.loop.ii.count", i32 4}
 template <int A>
-void boo() {
+void ii() {
   int a[10];
+  // CHECK: ![[MD_II]] = distinct !{![[MD_II]], ![[MD_ii:[0-9]+]]}
+  // CHECK-NEXT: ![[MD_ii]] = !{!"llvm.loop.ii.count", i32 4}
   [[intelfpga::ii(A)]]
   for (int i = 0; i != 10; ++i)
     a[i] = 0;
-  // CHECK: ![[MD_D]] = distinct !{![[MD_D]], ![[MD_ii_3:[0-9]+]]}
-  // CHECK-NEXT: ![[MD_ii_3]] = !{!"llvm.loop.ii.count", i32 8}
+  // CHECK: ![[MD_II_2]] = distinct !{![[MD_II_2]], ![[MD_ii_2:[0-9]+]]}
```

Comment:
`_2` has an alphabetic feel to it as well. Not sure if I can suggest anything better than `_template`, though

Reply:
I think `_2` is okay as long as we usually update test with new functions, not with new test cases in function. Anyway it will be possible to just add new test case after the last one, so numeric order will not be ruined